csrc/optimization/remove_empty.cpp

```diff
@@ -260,8 +260,16 @@ class EmptyTensorRemover : public DeadCodeRemover {
     if (non_empty_inputs.size() != cop->inputs().size()) {
       // Replace this op with a new cat op
       auto old_tv = cop->outputs()[0]->as<TensorView>();
-      // NOTE: cat() will translate to set() if non_empty_inputs.size() == 1
-      auto new_tv = cat(non_empty_inputs, dim);
+      // NOTE: cat() will translate to set() if non_empty_inputs.size() == 1.
+      // Also note that unless we're careful this call to cat() might result in
+      // symbolic axis, since the inputs have potentially symbolic extents in
+      // the cat dimension. However, since we have already undergone
+      // concretization at this point, we can trust that the original IterType,
```

Comment:
The input extents are symbolic (e.g. `i0`) or they are `zeroVal()`, but since we use the `cat` command here to do the replacement, we need some way to tell it to not use `IterType::Symbolic` and trust us at this point that the output will be `Iteration`.

Reply:
Yes exactly. It should be more clear in the comment probably.

Reply:
Updated the comment.